### PR TITLE
Adding revision restore and delete implementation

### PR DESCRIPTION
**Pull request by CrowleyRajapakse**
## Issues
wso2/product-apim#9406

## Implementation
Revision Restore 
Revision Delete

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java [hunk 1]
```diff
@@ -3959,47 +3976,54 @@ public Response getAPIRevisionDeployments(String apiId, MessageContext messageCo
...
     @Override
-    public Response restoreAPIRevision(String apiId, String apiRevisionId, MessageContext messageContext) {
-        // remove errorObject and add implementation code!
-        ErrorDTO errorObject = new ErrorDTO();
-        Response.Status status = Response.Status.NOT_IMPLEMENTED;
-        errorObject.setCode((long) status.getStatusCode());
-        errorObject.setMessage(status.toString());
-        errorObject.setDescription("The requested resource has not been implemented");
-        return Response.status(status).entity(errorObject).build();
+    public Response restoreAPIRevision(String apiId, String revisionId, MessageContext messageContext) {
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java:4017]: throws APIManagementException

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java [hunk 2]
```diff
@@ -3959,47 +3976,54 @@ public Response getAPIRevisionDeployments(String apiId, MessageContext messageCo
...
+        try {
+            APIProvider apiProvider = RestApiCommonUtil.getLoggedInUserProvider();
+            apiProvider.restoreAPIRevision(apiId, revisionId);
+            APIDTO apiToReturn = getAPIByID(apiId);
+            return Response.ok().entity(apiToReturn).build();
+        } catch (APIManagementException e) {
+            String errorMessage = "Error while restoring API Revision for API : " + apiId + ". Error: " + e;
+            RestApiUtil.handleInternalServerError(errorMessage, e, log);
+        }
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java:4026]: doesn't need

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml [hunk 3]
```diff
@@ -7254,10 +7266,10 @@ components:
     APIRevision:
       title: API Info object with basic API details
       properties:
+        key:
+          type: string
+          example: REVISION 1
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml:7271]: typo?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java (hunk 1 above)
Comment (CrowleyRajapakse) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java]: resolved

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java (hunk 2 above)
Comment (CrowleyRajapakse) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java]: resolved

**Review by malinthaprasan — APPROVED** [review]

**Pull request merged**